mlflow/projects/databricks.py

```diff
@@ -27,6 +47,112 @@ def _jobs_runs_submit(req_body_json):
+    export PATH=$PATH:$DB_HOME/python/bin:/$DB_HOME/conda/bin &&
+    mlflow --version &&
+    # Make local directories in the container into which to copy/extract the tarred project
+    mkdir -p {0} && mkdir -p {1} &&
+    # Rsync from DBFS FUSE to avoid copying archive into local filesystem if it already exists
+    rsync -a -v --ignore-existing {2} {0} &&
+    # Extract project into a temporary directory. We don't extract directly into the desired
+    # directory as tar extraction isn't guaranteed to be atomic
+    cd $(mktemp -d) &&
+    tar -xzvf {3} &&
+    # Atomically move the extracted project into the desired directory
+    mv -T {4} {5} &&
+    {6}
+    """.format(DB_TARFILE_BASE, DB_PROJECTS_BASE, dbfs_fuse_tar_uri, container_tar_path,
+               DB_TARFILE_ARCHIVE_NAME, project_dir, mlflow_run_cmd))
+    return ["bash", "-c", shell_command]
+
+
+def _check_databricks_cli_configured():
+    cfg_file = os.path.join(os.path.expanduser("~"), ".databrickscfg")
+    try:
+        process.exec_cmd(["databricks", "--version"])
+    except process.ShellCommandException:
+        raise ExecutionException(
+            "Could not find Databricks CLI on PATH. Please install and configure the Databricks "
+            "CLI as described in https://github.com/databricks/databricks-cli")
+    if not os.path.exists(cfg_file):
+        raise ExecutionException("Could not find profile for Databricks CLI in %s. Make sure the "
+                                 "the Databricks CLI is installed and that credentials have been "
+                                 "configured as described in "
+                                 "https://github.com/databricks/databricks-cli" % cfg_file)
+
+
+def _upload_to_dbfs(src_path, dbfs_uri):
+    """
+    Uploads the file at `src_path` to the specified DBFS URI within the Databricks workspace
+    corresponding to the default Databricks CLI profile.
+    """
+    process.exec_cmd(cmd=["databricks", "fs", "cp", src_path, dbfs_uri])
+
+
+def _dbfs_path_exists(dbfs_uri):
+    """
+    Returns True if the passed-in path exists in DBFS for the workspace corresponding to the
+    default Databricks CLI profile.
+    """
+    dbfs_path = _parse_dbfs_uri_path(dbfs_uri)
+    res = rest_utils.databricks_api_request(
+        endpoint="dbfs/get-status", method="GET", params={"path": dbfs_path},
+        verify_success=False)
+    # If request fails with a RESOURCE_DOES_NOT_EXIST error, the file does not exist on DBFS
+    error_code_field = "error_code"
+    if error_code_field in res:
+        if res[error_code_field] == "RESOURCE_DOES_NOT_EXIST":
+            return False
+        raise ExecutionException("Got unexpected error response when checking whether file %s "
+                                 "exists DBFS: %s" % res)
+    return True
+
+
+def _upload_project_to_dbfs(project_dir, experiment_id):
+    """
+    Tars a project directory into an archive in a temp dir, returning the path to the
+    tarball.
```

Comment:
returning the path to the tarball in DBFS. Can we also give an example of this? E.g., is it prepended w/`dbfs:/`?